File: samples/java-azurefunction/src/main/java/com/function/InvokeOutputBinding.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package main.java.com.function;
import com.microsoft.azure.functions.ExecutionContext;
import com.microsoft.azure.functions.HttpMethod;
import com.microsoft.azure.functions.HttpRequestMessage;
import com.microsoft.azure.functions.HttpResponseMessage;
import com.microsoft.azure.functions.HttpStatus;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;
import com.microsoft.azure.functions.dapr.annotation.DaprInvokeOutput;
import com.microsoft.azure.functions.dapr.annotation.DaprStateOutput;
import com.microsoft.azure.functions.OutputBinding;

import java.util.Optional;

/**
 * This function uses Dapr Invoke Output Binding to invoke another Dapr enabled function.
 * This function can be invoked by http trigger.
 */
public class InvokeOutputBinding {
    /**
     * This function listens at endpoint "/api/invoke/{appId}/{methodName}". Curl command to invoke it:
     * curl -X POST \                                                         ─╯
            http://localhost:7071/api/invoke/functionapp/CreateNewOrder \
            -H 'Content-Type: application/json' \
            -d '{
                "body":{
                    "value": {
                        "orderId": "45"
                    }
                }
            }'
     */
    @FunctionName("InvokeOutputBinding")
    public String run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "invoke/{appId}/{methodName}")
                HttpRequestMessage<Optional<String>> request,
            @DaprInvokeOutput(
                name = "payload",
                appId = "{appId}", 
                methodName = "{methodName}", 
                httpVerb = "post")
            OutputBinding<String> payload,
            final ExecutionContext context) {
        context.getLogger().info("Java HTTP trigger processed a request.");

        // Parse query parameter
        final String query = request.getQueryParameters().get("name");
        final String body = request.getBody().orElse(query);

        //request body must be passed this way "{\"body\":{\"value\":{\"key\":\"some value\"}}}" to use the InvokeOutputBinding, all the data must be enclosed in body property.
        String jsoString = String.format("{\"body\":\"%s\"}", body);

        payload.setValue(jsoString);

        return jsoString;
    }
}

File: samples/python-v2-azurefunctions/invoke_output_binding.py
import logging 
import azure.functions as func 

invokeOutputBinding = func.DaprBlueprint()

# Dapr invoke output binding with http trigger
@invokeOutputBinding.function_name(name="InvokeOutputBinding")
@invokeOutputBinding.route(route="invoke/{appId}/{methodName}", auth_level=func.AuthLevel.ANONYMOUS)
@invokeOutputBinding.dapr_invoke_output(arg_name = "payload", app_id = "{appId}", method_name = "{methodName}", http_verb = "post")
def main(req: func.HttpRequest, payload: func.Out[str] ) -> str:
    # request body must be passed this way "{\"body\":{\"value\":{\"key\":\"some value\"}}}" to use the InvokeOutputBinding, all the data must be enclosed in body property.
    logging.info('Python function processed a InvokeOutputBinding request from the Dapr Runtime.')
    
    body = req.get_body()
    logging.info(body)
    if body is not None:
        payload.set(body)
    else:
        logging.info('req body is none')
    return 'ok'